PhysicsTools/NanoAOD/python/triggerObjects_cff.py

```diff
@@ -255,12 +257,26 @@
 for sel in selections2016:
     if sel.name=='Muon':
         sel.sel = cms.string("type(83) && pt > 5 && (coll('hlt*L3MuonCandidates') || coll('hlt*TkMuonCands') || coll('hlt*TrkMuonCands'))")
-        sel.qualityBits = cms.string("filter('*RelTrkIso*Filtered0p4') + 2*filter('hltL3cr*IsoFiltered0p09') + 4*filter('*OverlapFilter*IsoMu*PFTau*') + 8*filter('hltL3f*IsoFiltered0p09') + 1024*max(filter('hltL3fL1sMu*L3Filtered50*'),filter('hltL3fL1sMu*TkFiltered50*'))")
-        sel.qualityBitsDoc = cms.string("1 = TrkIsoVVL, 2 = Iso, 4 = OverlapFilter PFTau, 8 = IsoTkMu, 1024 = 1mu (Mu50)")
+        sel.qualityBits = cms.VPSet(
+            mksel("filter('*RelTrkIso*Filtered0p4')","TrkIsoVVL"),
+            mksel("filter('hltL3cr*IsoFiltered0p09')","Iso"),
+            mksel("filter('*OverlapFilter*IsoMu*PFTau*')","OverlapFilter PFTau"),
+            mksel("filter('hltL3f*IsoFiltered0p09')","IsoTkMu"),
+            mksel(["hltL3fL1sMu*L3Filtered50*","hltL3fL1sMu*TkFiltered50*"],"1mu (Mu50)")
+        )
     elif sel.name=='Tau':
         sel.sel = cms.string("type(84) && pt > 5 && coll('*Tau*') && ( filter('*LooseIso*') || filter('*MediumIso*') || filter('*MediumComb*Iso*') || filter('hltL2TauIsoFilter') || filter('*OverlapFilter*IsoMu*') || filter('*OverlapFilter*IsoEle*') || filter('*L1HLTMatched*') || filter('*Dz02*') )")
-        sel.qualityBits = cms.string("(filter('*LooseIso*')-filter('*VLooseIso*'))+2*filter('*Medium*Iso*')+4*filter('*VLooseIso*')+8*0+16*filter('hltL2TauIsoFilter')+32*filter('*OverlapFilter*IsoMu*')+64*filter('*OverlapFilter*IsoEle*')+128*filter('*L1HLTMatched*')+256*filter('*Dz02*')")
-        sel.qualityBitsDoc = cms.string("1 = LooseIso, 2 = Medium(Comb)Iso, 4 = VLooseIso, 8 = None, 16 = L2p5 pixel iso, 32 = OverlapFilter IsoMu, 64 = OverlapFilter IsoEle, 128 = L1-HLT matched, 256 = Dz")
+        sel.qualityBits = cms.VPSet(
+            mksel("(filter('*LooseIso*')-filter('*VLooseIso*'))","LooseIso"),
```

Comment:
This is strange, why the `-`?

Reply:
everything that is `*LooseIso*`but not `*VLooseIso*`

Reply:
Sure, but since `VLooseIso` is stored anyway two bits down, the user could recover that logic, I don't see the point of having it here...

Reply:
indeed, I guess this was intended to be directly consummed, instead of having to do "one but not the other" from nano ; we could change that, though it does not cost anything like this